src/Features/Core/Portable/Completion/Providers/AbstractAwaitCompletionProvider.cs

```diff
@@ -137,9 +138,9 @@ public sealed override async Task ProvideCompletionsAsync(CompletionContext cont
             return;
 
             static CompletionItem CreateCompletionItem(
-                ImmutableDictionary<string, string> completionProperties, string displayText, string filterText, string tooltip, bool isComplexTextEdit)
+                ArrayBuilder<KeyValuePair<string, string>> completionProperties, string displayText, string filterText, string tooltip, bool isComplexTextEdit)
             {
-                var appendConfigureAwait = completionProperties.ContainsKey(AppendConfigureAwait);
+                var appendConfigureAwait = completionProperties.Any(static (kvp, value) => kvp.Key == value, AppendConfigureAwait);
```

Comment:
nit: the caller already knows about this, consider passing in a `bool appendConfigureAwait` instead.

Reply:
nice, I like that